pkg/addonmanager/controllers/registration/controller.go

```diff
@@ -155,23 +162,29 @@ func (c *addonRegistrationController) sync(ctx context.Context, syncCtx factory.
 		_, err = addonPatcher.PatchStatus(ctx, managedClusterAddonCopy, managedClusterAddonCopy.Status, managedClusterAddon.Status)
 		return err
 	}
-	configs := registrationOption.CSRConfigurations(managedCluster)
-
-	managedClusterAddonCopy.Status.Registrations = configs
-
-	managedClusterAddonCopy.Status.Namespace = registrationOption.Namespace
-	if len(managedClusterAddonCopy.Spec.InstallNamespace) > 0 {
-		managedClusterAddonCopy.Status.Namespace = managedClusterAddonCopy.Spec.InstallNamespace
-	}
 
+	var agentInstallNamespace string
 	if registrationOption.AgentInstallNamespace != nil {
-		ns, err := registrationOption.AgentInstallNamespace(managedClusterAddonCopy)
+		agentInstallNamespace, err = registrationOption.AgentInstallNamespace(managedClusterAddonCopy)
 		if err != nil {
+			if _, patchErr := addonPatcher.PatchStatus(
```

Comment:
so the status.SupportedConfigs already patched above, we do not need patch again here?

Reply:
@zhujian7 yes, I think so, will remove this. Could you also help confirm, when failed to get `registrationOption.AgentInstallNamespace(managedClusterAddonCopy)`, we don't need to set `Status.Registrations` as well, right?